app/kumactl/cmd/apply/apply.go

```diff
@@ -37,8 +46,28 @@ func NewApplyCmd(pctx *kumactl_cmd.RootContext) *cobra.Command {
 
 			if ctx.args.file == "" || ctx.args.file == "-" {
 				b, err = ioutil.ReadAll(cmd.InOrStdin())
+				if err != nil {
+					return err
+				}
 			} else {
-				b, err = ioutil.ReadFile(ctx.args.file)
+				if strings.HasPrefix(ctx.args.file, "http") {
```

Comment:
Might be better to use `net/url` `ParseRequestURI` instead, which returns an error if the URL is not valid.

Reply:
The problem with that is it will also parse stuff like `file://`
https://play.golang.org/p/zFii42x2kkU

We explicitly want `http://` or `https://`. But it's a good point that it will break files that begins with http. We can change it to accept prefix `http://` or `https://` or try to parse URI and see if schema is `http` or `https`. If think either option is fine.

Reply:
@devadvocado 
getting back to this comment. I think this should be
```
if strings.HasPrefix(ctx.args.file, "http://") || strings.HasPrefix(ctx.args.file, "https://") {
```
also I'd extract content of this if to some function: ex. `func readFromUrl(url string) ([]byte, error)`.

Reply:
done